torch/csrc/utils/future.h

```diff
@@ -156,6 +154,25 @@ class TORCH_API Future final {
     }
   }
 
+  void markCompletedInternal(T value,
+      std::unique_lock<std::mutex>& lock) {
+    TORCH_CHECK(!completed_);
+    // Set value first as completed_ is accessed without lock
```

Comment:
What does this comment mean? Looks like the lock is always held before entering this method, no?

Reply:
It seems incorrect bc we have the lock - the comment was just a remnant of the old markCompleted function, but I'll remove